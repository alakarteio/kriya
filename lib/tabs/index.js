'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _tabs = require('./tabs.container');

Object.defineProperty(exports, 'default', {
  enumerable: true,
  get: function get() {
    return _interopRequireDefault(_tabs).default;
  }
});

var _tab = require('./tab');

Object.defineProperty(exports, 'Tab', {
  enumerable: true,
  get: function get() {
    return _tab.Tab;
  }
});

var _redux = require('./redux');

Object.keys(_redux).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function get() {
      return _redux[key];
    }
  });
});
Object.defineProperty(exports, 'reducer', {
  enumerable: true,
  get: function get() {
    return _interopRequireDefault(_redux).default;
  }
});

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }