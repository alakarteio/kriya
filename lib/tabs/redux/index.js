'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _tabs = require('./tabs');

Object.defineProperty(exports, 'default', {
  enumerable: true,
  get: function get() {
    return _interopRequireDefault(_tabs).default;
  }
});

var _tabs2 = require('./tabs.actions');

Object.keys(_tabs2).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function get() {
      return _tabs2[key];
    }
  });
});

var _tabs3 = require('./tabs.selectors');

Object.keys(_tabs3).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function get() {
      return _tabs3[key];
    }
  });
});

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }