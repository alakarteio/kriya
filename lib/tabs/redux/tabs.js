'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.initAction = exports.initState = undefined;

var _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; };

var _tabs = require('./tabs.actions');

function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }

function _toConsumableArray(arr) { if (Array.isArray(arr)) { for (var i = 0, arr2 = Array(arr.length); i < arr.length; i++) { arr2[i] = arr[i]; } return arr2; } else { return Array.from(arr); } }

var initState = exports.initState = { data: {}, keys: [], activated: undefined };
var initAction = exports.initAction = { type: 'UNKNOWN' };

exports.default = function () {
  var state = arguments.length > 0 && arguments[0] !== undefined ? arguments[0] : initState;

  var _ref = arguments.length > 1 && arguments[1] !== undefined ? arguments[1] : initAction,
      type = _ref.type,
      payload = _ref.payload;

  switch (type) {
    case _tabs.ADD_TAB:
      return _extends({}, state, {
        keys: [].concat(_toConsumableArray(state.keys), [payload.code]),
        data: _extends({}, state.data, _defineProperty({}, payload.code, payload))
      });
    case _tabs.ACTIVATE_TAB:
      return _extends({}, state, { activated: payload });
    case _tabs.RESET_TABS:
      return initState;
    default:
      return state;
  }
};