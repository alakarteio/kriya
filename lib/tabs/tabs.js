'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _propTypes = require('prop-types');

var _propTypes2 = _interopRequireDefault(_propTypes);

var _recompose = require('recompose');

var _classnames = require('classnames');

var _classnames2 = _interopRequireDefault(_classnames);

var _tabsStyles = require('../../src/tabs/tabs.styles.scss');

var _tabsStyles2 = _interopRequireDefault(_tabsStyles);

var _tab = require('./tab');

var _tab2 = _interopRequireDefault(_tab);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var Tabs = function Tabs(_ref) {
  var style = _ref.style,
      className = _ref.className,
      children = _ref.children,
      tabs = _ref.tabs;

  var classes = (0, _classnames2.default)(_tabsStyles2.default.tabs, className);

  return _react2.default.createElement(
    'div',
    { style: style, className: classes },
    tabs ? tabs.map(function (code) {
      return _react2.default.createElement(_tab2.default, { key: code, code: code });
    }) : children
  );
};

Tabs.propTypes = {
  style: _propTypes2.default.object,
  className: _propTypes2.default.string,
  children: _propTypes2.default.node,
  tabs: _propTypes2.default.arrayOf(_propTypes2.default.string)
};

Tabs.defaultProps = {
  style: {},
  className: '',
  children: undefined,
  tabs: undefined
};

exports.default = (0, _recompose.onlyUpdateForPropTypes)(Tabs);