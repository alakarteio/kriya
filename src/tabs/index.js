export { default } from './tabs.container'
export { Tab } from './tab'
