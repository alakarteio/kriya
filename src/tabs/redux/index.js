export { default } from './tabs'
export * from './tabs.actions'
export * from './tabs.selectors'
