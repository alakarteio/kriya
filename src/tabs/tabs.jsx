import React from 'react'
import PropTypes from 'prop-types'
import { onlyUpdateForPropTypes } from 'recompose'
import classnames from 'classnames'
import styles from '../../src/tabs/tabs.styles.scss'
import Tab from './tab'

const Tabs = ({ style, className, children, tabs }) => {
  const classes = classnames(
    styles.tabs,
    className,
  )

  return (
    <div style={style} className={classes}>
      {tabs ?
        tabs.map(code => (
          <Tab key={code} code={code} />
        ))
        : children
      }
    </div>
  )
}

Tabs.propTypes = {
  style: PropTypes.object,
  className: PropTypes.string,
  children: PropTypes.node,
  tabs: PropTypes.arrayOf(PropTypes.string),
}

Tabs.defaultProps = {
  style: {},
  className: '',
  children: undefined,
  tabs: undefined,
}

export default onlyUpdateForPropTypes(Tabs)
