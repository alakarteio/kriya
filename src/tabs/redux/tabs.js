import { ADD_TAB, ACTIVATE_TAB, RESET_TABS } from './tabs.actions'

export const initState = { data: {}, keys: [], activated: undefined }
export const initAction = { type: 'UNKNOWN' }

export default (state = initState, { type, payload } = initAction) => {
  switch (type) {
    case ADD_TAB: return {
      ...state,
      keys: [...state.keys, payload.code],
      data: { ...state.data, [payload.code]: payload },
    }
    case ACTIVATE_TAB: return { ...state, activated: payload }
    case RESET_TABS: return initState
    default: return state
  }
}
