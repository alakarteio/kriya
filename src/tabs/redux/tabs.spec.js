/* eslint-disable comma-dangle */
/* eslint-env jest */
import reducer from './tabs'
import { getActivated, getData, getTab, getTabs, isActivated } from './tabs.selectors'
import { activate, add, reset } from './tabs.actions'

const snap = object => expect(object).toMatchSnapshot()

describe('tabs/redux', () => {
  const state = {
    ui: {
      tabs: {
        keys: ['SEARCH', 'FIND'],
        data: {
          SEARCH: { label: 'Search' },
          FIND: { label: 'Find' },
        },
        activated: 'FIND',
      },
    },
  }

  describe('selectors', () => {
    it('should return tabs sub-reducer', () => snap(getTabs(state)))
    it('should return data', () => snap(getData(state)))
    it('should return tab', () => snap(getTab(state)('SEARCH')))
    it('should return tab -notfound-', () => snap(getTab(state)('LOOK')))
    it('should return activated tabs', () => snap(getActivated(state)))
    it('should return that FIND is activated', () => snap(isActivated(state)('FIND')))
  })

  describe('reducer', () => {
    const snapReduce = action => snap(reducer(getTabs(state), action))

    it('should initialize', () => snap(reducer()))
    it('should handle unknown action', () => expect(reducer(getTabs(state), { type: 'YO' })).toBe(getTabs(state)))
    it('should activate SEARCH tab', () => snapReduce(activate('SEARCH')))
    it('should add LOOK tab', () => snapReduce(add({ code: 'LOOK', label: 'Look up !' })))
    it('should reset tabs', () => snapReduce(reset()))
  })
})

/* eslint-enable comma-dangle */
