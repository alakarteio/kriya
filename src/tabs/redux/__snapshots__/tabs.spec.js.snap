// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tabs/redux reducer should activate SEARCH tab 1`] = `
Object {
  "activated": "SEARCH",
  "data": Object {
    "FIND": Object {
      "label": "Find",
    },
    "SEARCH": Object {
      "label": "Search",
    },
  },
  "keys": Array [
    "SEARCH",
    "FIND",
  ],
}
`;

exports[`tabs/redux reducer should add LOOK tab 1`] = `
Object {
  "activated": "FIND",
  "data": Object {
    "FIND": Object {
      "label": "Find",
    },
    "LOOK": Object {
      "code": "LOOK",
      "label": "Look up !",
    },
    "SEARCH": Object {
      "label": "Search",
    },
  },
  "keys": Array [
    "SEARCH",
    "FIND",
    "LOOK",
  ],
}
`;

exports[`tabs/redux reducer should initialize 1`] = `
Object {
  "activated": undefined,
  "data": Object {},
  "keys": Array [],
}
`;

exports[`tabs/redux reducer should reset tabs 1`] = `
Object {
  "activated": undefined,
  "data": Object {},
  "keys": Array [],
}
`;

exports[`tabs/redux selectors should return activated tabs 1`] = `"FIND"`;

exports[`tabs/redux selectors should return data 1`] = `
Object {
  "FIND": Object {
    "label": "Find",
  },
  "SEARCH": Object {
    "label": "Search",
  },
}
`;

exports[`tabs/redux selectors should return tab -notfound- 1`] = `undefined`;

exports[`tabs/redux selectors should return tab 1`] = `
Object {
  "label": "Search",
}
`;

exports[`tabs/redux selectors should return tabs sub-reducer 1`] = `
Object {
  "activated": "FIND",
  "data": Object {
    "FIND": Object {
      "label": "Find",
    },
    "SEARCH": Object {
      "label": "Search",
    },
  },
  "keys": Array [
    "SEARCH",
    "FIND",
  ],
}
`;

exports[`tabs/redux selectors should return that FIND is activated 1`] = `true`;
